sci-mathematics/calc/calc-2.12.5.3.recipe

```diff
@@ -0,0 +1,149 @@
+PATCHES="calc-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86_gcc2 x86"
+
+PROVIDES="
+	calc$secondaryArchSuffix = $portVersion
+	cmd:calc$secondaryArchSuffix = $portVersion
+	cmd:cscript$secondaryArchSuffix = $portVersion
+	lib:libcalc$secondaryArchSuffix
+	lib:libcustcalc$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libreadline$secondaryArchSuffix
+	lib:libhistory$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	calc_data == $portVersion
+	"
+
+if [ -z "$secondaryArchSuffix" ]; then
+	SUMMARY_data="Standard resource files for calc"
+	ARCHITECTURES_data="any"
+	PROVIDES_data="
+		calc_data = $portVersion
+		"
+	REQUIRES_data=""
+fi
+
+PROVIDES_devel="
+	calc${secondaryArchSuffix}_devel = $portVersion
+	devel:libcalc$secondaryArchSuffix
+	devel:libcustcalc$secondaryArchSuffix
+	"
+REQUIRES_devel="
+	calc$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libreadline$secondaryArchSuffix
+	devel:libhistory$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:awk
+	cmd:sed
+	cmd:diff
+	cmd:egrep
+	cmd:ctags
+	cmd:xargs
+	cmd:cmp
+	cmd:touch
+	cmd:less
+	cmd:strip
+	"
+
+BUILD()
+{
+	make target=Haiku \
+		INCDIR="$includeDir" \
+		LIBDIR=$libDir \
+		BINDIR=$binDir \
+		SCRIPTDIR=$libDir/calc/cscript \
+		CALC_SHAREDIR=$dataDir/calc \
+		MANDIR=$manDir/man1 \
+		BLD_TYPE=calc-dynamic-only \
+		HAVE_UNISTD_H=YES \
+		HAVE_FPOS_POS=1 \
+		HAVE_UID_T=1 \
+		HAVE_USTAT=1 \
+		EXTRA_CFLAGS="-DUSE_TERMIOS" \
+	 	USE_READLINE=-DUSE_READLINE \
+		READLINE_LIB=-lreadline \
+		READLINE_EXTRAS="-lhistory -lncurses" \
+		LN="ln -s" \
+		LDCONFIG= \
+		WNO_ERROR_LONG_LONG=
+}
+
+INSTALL()
+{
+	make install \
+		target=Haiku \
+		INCDIR="$includeDir" \
+		LIBDIR=$libDir \
+		BINDIR=$binDir \
+		SCRIPTDIR=$libDir/calc/cscript \
+		CALC_SHAREDIR=$dataDir/calc \
+		MANDIR=$manDir/man1 \
+		BLD_TYPE=calc-dynamic-only \
+		HAVE_UNISTD_H=YES \
+		HAVE_FPOS_POS=1 \
+		HAVE_UID_T=1 \
+		HAVE_USTAT=1 \
+		EXTRA_CFLAGS="-DUSE_TERMIOS" \
+	 	USE_READLINE=-DUSE_READLINE \
+		READLINE_LIB=-lreadline \
+		READLINE_EXTRAS="-lhistory -lncurses" \
+		LN="ln -s" \
+		LDCONFIG= \
+		WNO_ERROR_LONG_LONG=
+
+	prepareInstalledDevelLibs libcalc libcustcalc
+	fixPkgconfig
+
+	if [ -z "$secondaryArchSuffix" ]; then
+		# Shorten some paths in the man page.
+		sed -i \
+			-e "s|#!$binDir/calc\>|#!/$relativeBinDir/calc|" \
+			-e "s|$prefix/|/system/|" \
+			$manDir/man1/calc.1
+
+		# Some files in $dataDir/calc/help/ have code snippets with
+		# long shebangs that we can shorten. Currently, these files are
+		# cscript, full, libcalc and usage.
+		sed -i -e "1 s|$binDir/calc |/$relativeBinDir/calc |" \
+			$dataDir/calc/help/*
+
+		packageEntries data \
+			$dataDir/calc
+	else
+		rm -rf $dataDir/calc $manDir
+	fi
+
```

Comment:
@fbrosson Some of them that i patched in makefile ex: HAVE_STDLIB_H instead commandline because there's bug it wont apply to the makefile when i patch it with commandline.